### PR TITLE
refactor: Extract "HttpStats" utility class

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Wrap HTTP statistics into the helper class
- Add median and max data

## Context

- Ref: #27035

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/http/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable import/order */
```
Comment (viceice) [inline lib/util/http/index.ts:1]: why?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/util/http/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/util/http/index.ts:1]: Oops, wrong fix suggestion

**Comment by rarkins** [comment]
Is the output structure identical? Adding @nabeelsaabna because he parses some of these stats in our backend

**Review by viceice — DISMISSED** [review]

**Comment by rarkins** [comment]
@zharinov do you have a before/after example?

**Comment by zharinov** [comment]
No, it's not identical

**Comment by zharinov** [comment]
And previous ones are different too, BTW

**Comment by zharinov** [comment]
```
DEBUG: Package cache statistics (repository=renovate-testing/26315-extract-rust-versions-for-releases)
       "get": {"count": 2, "avgMs": 2, "medianMs": 2, "maxMs": 2, "totalMs": 3},
       "set": {"count": 2, "avgMs": 3, "medianMs": 4, "maxMs": 4, "totalMs": 6}
DEBUG: HTTP stats (repository=renovate-testing/26315-extract-rust-versions-for-releases)
       "urlCounts": {
         "https://api.github.com/graphql (POST, 200)": 1,
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/branches/main/protection (GET, 404)": 1,
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/contents/renovate.json (GET, 304)": 1,
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/issues/3 (GET, 304)": 1,
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/pulls (GET, 304)": 1,
         "https://crates.io/api/v1/crates/log (GET, 200)": 1,
         "https://raw.githubusercontent.com/rust-lang/crates.io-index/master/3/l/log (GET, 200)": 1
       },
       "statsByHost": {
         "api.github.com": {
           "count": 5,
           "reqAvgMs": 278,
           "reqMedianMs": 269,
           "reqMaxMs": 373,
           "queueAvgMs": 0,
           "queueMedianMs": 0,
           "queueMaxMs": 1
         },
         "crates.io": {
           "count": 1,
           "reqAvgMs": 657,
           "reqMedianMs": 657,
           "reqMaxMs": 657,
           "queueAvgMs": 1,
           "queueMedianMs": 1,
           "queueMaxMs": 1
         },
         "raw.githubusercontent.com": {
           "count": 1,
           "reqAvgMs": 301,
           "reqMedianMs": 301,
           "reqMaxMs": 301,
           "queueAvgMs": 0,
           "queueMedianMs": 0,
           "queueMaxMs": 0
         }
       },
       "totalRequests": 7
DEBUG: Lookup statistics (repository=renovate-testing/26315-extract-rust-versions-for-releases)
       "crate": {"count": 1, "avgMs": 1010, "medianMs": 1010, "maxMs": 1010, "totalMs": 1010}
```

**Comment by rarkins** [comment]
Looks ok, I will wait for @nabeelsaabna to confirm we can use it or adapt to it

**Comment by nabeelsaabna** [comment]
I see mainly name changes that we can update easily 👍 

`urls` -> `urlCounts`
`hostStats` -> `statsByHost`

old log example
```
{
	"level": 20,
	"urls": {
		"https://api.github.com/graphql (POST,200)": 1
	},
	"hostStats": {
		"api.github.com": {
			"requestCount": 1,
			"requestAvgMs": 93,
			"queueAvgMs": 0
		}
	},
	"totalRequests": 3,
	"msg": "http statistics"
}
```

**Comment by nabeelsaabna** [comment]
Side note: for using this message in graphs/gathering metrics and trends it would be more useful to extract the `(POST,200)` from the URL key
e.g.
```
{
	"urls": {
		"https://api.github.com/graphql": {
			"POST": {
				"200": 1
				"404": 3
			}
		},
		"https://crates.io/api/v1/crates/log": {
			"GET": {
				"200": 1
			}
		}
	}
}
```

**Comment by zharinov** [comment]
Please decide what changes to make, I'll do it later today. Now I have to go and pass Spanish exam, see you.

**Review by nabeelsaabna — DISMISSED** [review]
Approving naming/structure changes

**Comment by nabeelsaabna** [comment]
> Please decide what changes to make, I'll do it later today. Now I have to go and pass Spanish exam, see you.

no changes need in this PR

**Comment by zharinov** [comment]
Almost done adjusting the structure:

```
"urls": {
         "https://api.github.com/graphql": {"POST": {"200": 1}},
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/branches/main/protection": {
           "GET": {"404": 1}
         },
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/contents/renovate.json": {
           "GET": {"304": 1}
         },
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/issues/3": {
           "GET": {"304": 1}
         },
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/pulls": {
           "GET": {"304": 1}
         }
       }
```

Would you like to have it like this? Or never mind?

**Comment by zharinov** [comment]
The final log:

```
DEBUG: HTTP stats (repository=renovate-testing/26315-extract-rust-versions-for-releases)
       "urls": {
         "https://api.github.com/graphql": {"POST": {"200": 1}},
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/branches/main/protection": {
           "GET": {"404": 1}
         },
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/contents/renovate.json": {
           "GET": {"304": 1}
         },
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/issues/3": {
           "GET": {"304": 1}
         },
         "https://api.github.com/repos/renovate-testing/26315-extract-rust-versions-for-releases/pulls": {
           "GET": {"304": 1}
         }
       },
       "hosts": {
         "api.github.com": {
           "count": 5,
           "reqAvgMs": 259,
           "reqMedianMs": 248,
           "reqMaxMs": 308,
           "queueAvgMs": 0,
           "queueMedianMs": 0,
           "queueMaxMs": 0
         }
       },
       "requests": 5
```

Field names are `urls`, `hosts`, `requests`

**Comment by rarkins** [comment]
I think the get/post improvement is useful!

**Review by viceice — COMMENTED** [review]
needs prettier fix

**Review by viceice — APPROVED** [review]

**Pull request merged**